nexus/src/db/datastore/silo_group.rs

```diff
@@ -36,10 +36,30 @@ impl DataStore {
         opctx.authorize(authz::Action::CreateChild, authz_silo).await?;
 
         use db::schema::silo_group::dsl;
+
+        // Without the on_conflict below, simultaneous attempts to ensure the
+        // same SiloGroup will see one of them fail. This can occur if two users
+        // are logging in at the same time, and both are part of IdP groups that
+        // do not yet exist in our database.
+        //
+        // Currently there is a unique partial index on silo_group, which
```

Comment:
I thought the Diesel we're using did support this now.  Is that not right?

Reply:
That part of the comment refers to "diesel does not support creating queries that contain `ON CONFLICT (...) WHERE ... DO NOTHING`.

I looked at https://github.com/oxidecomputer/omicron/issues/1545 and saw that

```rust
.on_conflict((dsl::silo_id, dsl::external_id))
.filter_target(dsl::time_deleted.is_null())
.do_nothing()
```

was suggested, and it works! 643aadb3 changes this and removes the comment.